var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -29,6 +29,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     version("master", branch="master")
     version("develop", branch="develop")
 
+    version("4.5.00", sha256="cbfb742feeb9e649db9eca0394e6ca9a22aa017a1e6aab8576990772a0e3135b")
```

Comment:
Why do these not include a `url` field?

Reply:
They are built automatically based on the url of line 14. This way, `spack checksum` will find the correct tarball, not the one created by github.
I do not know how to check signature however.

Reply:
https://github.com/spack/spack/blob/16fd77f9da7e3865ed4e22f7a9bf1d3b0b048006/var/spack/repos/builtin/packages/kokkos/package.py#L19

spack uses this URL as a template for all the versions listed.

And if the template doesn't match - one can add in the URL explicitly - for that version].

BTW: one can have multiple templates via `url_for_version()` - so the ones with the URL listed explicitly can be simplified.

Reply:
What do you mean regarding checking the signature? You want to have it done automatically by spack?

Reply:
Previous releases are doing
https://github.com/spack/spack/blob/16fd77f9da7e3865ed4e22f7a9bf1d3b0b048006/var/spack/repos/builtin/packages/kokkos/package.py#L36-L40

Reply:
As you know these previous releases relied on the automatic artifact from github which has a different url that the new releases

Reply:
Perhaps an update using the following format:
```patch
diff --git a/var/spack/repos/builtin/packages/kokkos/package.py b/var/spack/repos/builtin/packages/kokkos/package.py
index 6b58c7298a..bcd93cf281 100644
--- a/var/spack/repos/builtin/packages/kokkos/package.py
+++ b/var/spack/repos/builtin/packages/kokkos/package.py
@@ -33,11 +33,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     version("4.4.01", sha256="3413f0cb39912128d91424ebd92e8832009e7eeaf6fa8da58e99b0d37860d972")
     version("4.4.00", sha256="0b46372f38c48aa088411ac1b7c173a5c90f0fdb69ab40271827688fc134f58b")
 
-    version(
-        "4.3.01",
-        sha256="5998b7c732664d6b5e219ccc445cd3077f0e3968b4be480c29cd194b4f45ec70",
-        url="https://github.com/kokkos/kokkos/archive/4.3.01.tar.gz",
-    )
+    version("4.3.01",sha256="5998b7c732664d6b5e219ccc445cd3077f0e3968b4be480c29cd194b4f45ec70")
     version(
         "4.3.00",
         sha256="53cf30d3b44dade51d48efefdaee7a6cf109a091b702a443a2eda63992e5fe0d",
@@ -381,6 +377,12 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     ]
     sanity_check_is_dir = ["bin", "include"]
 
+    def url_for_version(self, ver):
+        if self.spec.satisfies("@:4.3"):
+            return f"https://github.com/kokkos/kokkos/archive/{ver}.tar.gz"
+        else:
+            return f"https://github.com/kokkos/kokkos/releases/download/{ver}/kokkos-{ver}.tar.gz"
+
     @classmethod
     def get_microarch(cls, target):
         """Get the Kokkos microarch name for a Spack target (spec.target)."""
```

Reply:
If we did that we might want to remove the url on line 19 then to avoid having two mechanisms that could diverge in the future?

Reply:
This way would be the best and allows eliminating many url fields from older versions.

Reply:
> If we did that we might want to remove the url on line 19 then to avoid having two mechanisms that could diverge in the future?

I'm not sure whats recommended here. I think many pkgs keep this line [so we also have it in petsc/package.py]  - servers more of documentation, but I see some [like  blast2go/package.py] don't have it.

Reply:
I've proposed this in the past and it was rejected by the package maintainers.

Reply:
Yes, my argument still stands that it does not change anything one way or another. Functionally everything is the same. The only benefit of the proposed approach would come in the future if we decided to change once more the url.